snuba/query/parser/__init__.py

```diff
@@ -127,9 +134,25 @@ def build_selected_expressions(
 
     arrayjoin = body.get("arrayjoin")
     if arrayjoin:
-        array_join_expr: Optional[Expression] = parse_expression(body["arrayjoin"])
+        array_join_expr: Optional[Expression] = parse_expression(
+            body["arrayjoin"], dataset.get_abstract_columnset()
+        )
     else:
         array_join_expr = None
+        for select_expr in select_clause:
+            if isinstance(select_expr.expression, FunctionCall):
+                if select_expr.expression.function_name == "arrayJoin":
+                    if arrayjoin:
+                        raise ParsingException(
+                            "Only one arrayJoin(...) call is allowed in a query."
+                        )
+
+                    parameters = select_expr.expression.parameters
+                    if len(parameters) != 1 or not isinstance(parameters[0], Column):
+                        raise ParsingException(
+                            "arrayJoin(...) only accepts a single column as a parameter."
+                        )
```

Comment:
Have you verified this never happens today in production ?

Reply:
I'm not sure how to do that. Is there something in Kibana I should search for?

Reply:
I would send a quick PR before this to collect metrics. And ensure this metric is empty.

Reply:
Opened PR: https://github.com/getsentry/snuba/pull/1252

Reply:
OK so I've looked over the metrics, and no one has even used the arrayJoin function once, and all the requests have only been a single column. I'm going to say this is OK to merge.

Reply:
ok, thanks, could you please remove the metrics now that we have confirmed ?